core/trino-spi/src/main/java/io/trino/spi/predicate/ValueSet.java

```diff
@@ -161,4 +166,78 @@ default boolean contains(ValueSet other)
     String toString(ConnectorSession session, int limit);
 
     long getRetainedSizeInBytes();
+
+    /**
+     * Try to expand {@code valueSet} into a discrete set of (at most {@code limit}) objects.
+     * For example: [1, 5] can be expanded into {1, 2, 3, 4, 5}.
+     * If the data type is not supported or the expansion results in too many values, {@code Optional.empty()} is returned.
+     */
+    static Optional<Collection<Object>> tryExpandRanges(ValueSet valuesSet, int valuesLimit)
```

Comment:
remove `static`

Reply:
Not sure - if I remove it, the build will fail (IIUC):
```
[ERROR] .../ValueSet.java:[176,5] interface abstract methods cannot have body
```

Reply:
The operation takes ValueSet and is defined in ValueSet. Why not instance/interface method?

Reply:
🤦

Reply:
You're right - my bad...

Reply:
Pushed a (hopefully) better version in 7a8425f088.